src/agent/debuglet.ts

```diff
@@ -113,6 +113,34 @@ const formatBreakpoints = function(
           .join('\n');
 };
 
+/**
+ * CachedPromise stores a promise. This promise is initially null and can be
+ * assigned to a new promise when refresh is called. This promise will be
+ * resolved by calling member function resolve.
+ */
+export class CachedPromise {
+  private promise: Promise<void>|null;
+  private promiseResolve: () => void;
+
+  constructor() {
+    this.clear();
+  }
+  refresh(): void {
+    this.promise = new Promise<void>((resolve) => {
+      this.promiseResolve = resolve;
+    });
+  }
+  get(): Promise<void>|null {
```

Comment:
I recommend lazily initializing `this.promise` here if it's null rather than have a separate `refresh()` function for it. This way, you can ensure the return type of `get()` is non-null.

Reply:
I did this because sometime we don't want to refresh the promise in get. For example, if previous `breakpointFetch` promised not being resolved yet, we wanted to return that promise by calling `get()` instead of refreshing a new one.